src/commonHelper.ts

```diff
@@ -0,0 +1,62 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as strings from './strings';
+import * as Constants from './constants';
+import { CommandObserver } from './CommandObserver';
+import * as compareVersions from 'compare-versions';
+const regex = /(?<="Microsoft.DataTools.Schema.Tasks.PostgreSql.Sdk\/)(.*?)(?=")/;
+
+export async function checkProjectVersion(minRequiredSDK: string, maxRequiredSDK: string, projects: string[], commandObserver: CommandObserver) : Promise<string[]> {
+	let unsupportedProjectsMap: string[] = [];
+    for (let project of projects) {
+        let projectFileText = fs.readFileSync(project, 'utf8');
+            let sdkVersion;
+            if ((sdkVersion = regex.exec(projectFileText)) !== null) {
+				if (!(compareVersions.compare(sdkVersion[0], minRequiredSDK, '>=') && compareVersions.compare(sdkVersion[0], maxRequiredSDK, '<='))) {
+                    unsupportedProjectsMap.push(project);
+                }
+            }
+    }
+
+    if (unsupportedProjectsMap.length > 0) {
+		promptToUpdateVersion(unsupportedProjectsMap, maxRequiredSDK, commandObserver);
+	}
+
+	return Promise.resolve(unsupportedProjectsMap);
+}
+
+function promptToUpdateVersion(unsupportedProjects: string[], maxRequiredSDK: string, commandObserver: CommandObserver) {
+    let msg = Constants.unsupportedPostgreSQLSdkMessage;
+    let installItem = 'Update Projects';
+    vscode.window
+        .showErrorMessage(msg, installItem)
+        .then(
+            (item) => {
+                if (item === installItem) {
+
+                    UpdateProjects(unsupportedProjects, maxRequiredSDK, commandObserver);
+                }
+            }
+        );
+}
+
+function UpdateProjects(unsupportedProjects: string[], maxRequiredSDK: string, commandObserver: CommandObserver) {
```

Comment:
Any reason why some functions are starting with lower-case and some start with upper-case? I think lower-case is the convention for JS/TypeScript.

Reply:
This function was part of a class and I noticed somewhere that for them they use upper case.. Let me see what the convention is.